overlord/snapstate/conflict.go

```diff
@@ -166,6 +166,13 @@ func CheckChangeConflictMany(st *state.State, instanceNames []string, ignoreChan
 		if chg == nil || chg.Status().Ready() {
 			continue
 		}
+
+		// task doesn't conflict if it can't run
+		switch task.Status() {
```

Comment:
this is equivalent to task.Status().Ready()

the change seems ok to me but probably needs to be done as a separate PR with some associated tests

Reply:
I wondered about doing it as a separate PR but opted not to since it was only triggered in my branch. I'll split it off into a separate PR with some better testing then

Reply:
Now that I had another look at this, I think this change is wrong. The issue is just that two updates to the same snap can conflict which means that the second one is retried after 30s while the test expects the task runner to converge sooner than that. I had wondered if blocking the install of the "main" snap because the default provider is outdated was ok. It might be better to significantly shorten the retry period or simply not retry at all, to prevent the "soft" dependency of the content provider from blocking an install for 30 seconds